theano/gpuarray/opt.py

```diff
@@ -250,53 +260,69 @@ def apply(self, fgraph):
         # Iterating through inputs of graph
         for i in fgraph.inputs:
             if isinstance(i.type, tensor.TensorType):
```

Comment:
If the input is a sparse variable, then we still need this check, otherwise as_gpuarray_variable will return an error as he don't know how to move tot he GPU sparse variable. It know only how to handle Tensor variables.